fenicsxprecice/adapter_core.py

```diff
@@ -160,17 +120,24 @@ def convert_fenicsx_to_precice(fenicsx_function, local_ids):
         raise Exception("Cannot handle data type {}".format(type(fenicsx_function)))
 
     precice_data = []
-
-    if fenicsx_function.function_space().num_sub_spaces() > 0:
-        dims = fenicsx_function.function_space().num_sub_spaces()
-        sampled_data = fenicsx_function.compute_vertex_values().reshape([dims, -1])
-    else:
-        sampled_data = fenicsx_function.compute_vertex_values()
+    # sampled_data = fenicsx_function.x.array  # that works only for 1st order elements where dofs = grid points
+    # TODO begin dirty fix
+    x_mesh = fenicsx_function.function_space.mesh.geometry.x
+    x_dofs = fenicsx_function.function_space.tabulate_dof_coordinates()
+    mask = []  # where dof coordinate == mesh coordinate
+    for i in range(x_dofs.shape[0]):
+        for j in range(x_mesh.shape[0]):
+            if np.allclose(x_dofs[i, :], x_mesh[j, :], 1e-15):
+                mask.append(i)
+                break
+    sampled_data = fenicsx_function.x.array[mask]
+    # end dirty fix
```

Comment:
I know we already discussed this. Why is this dirty fix needed? I understand if for the current situation it is good enough, but I think it would be good to open an issue that explains what's happening here, why a dirty fix is necessary and how a clean alternative could look like. @PhilipHildebrand could you take care of opening the issue and put a link to the issue here into the comments? I think @arvedes and I can try to add some details in the issue then. (For the sake of this PR the dirty fix is sufficient)

Reply:
Sure, here is the link: [https://github.com/precice/fenicsx-adapter/issues/17](url)

Reply:
```suggestion
    # TODO begin dirty fix. See https://github.com/precice/fenicsx-adapter/issues/17 for details.
    x_mesh = fenicsx_function.function_space.mesh.geometry.x
    x_dofs = fenicsx_function.function_space.tabulate_dof_coordinates()
    mask = []  # where dof coordinate == mesh coordinate
    for i in range(x_dofs.shape[0]):
        for j in range(x_mesh.shape[0]):
            if np.allclose(x_dofs[i, :], x_mesh[j, :], 1e-15):
                mask.append(i)
                break
    sampled_data = fenicsx_function.x.array[mask]
    # end dirty fix
```